pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -623,6 +628,10 @@ func (s *discovery) getServices() (*[]model.Service, error) {
 
 	var services []model.Service
 	alivePids := make(map[int32]struct{}, len(pids))
+	_, _, pidToCid, err := s.containerProvider.GetContainers(1*time.Minute, nil)
```

Comment:
I guess it's safe to use pidToCid even in the case of err since all the other callers also do it.

Reply:
There are two things to note on the subject:
- `GetContainers` actually never returns an error in practice, it is always nil. I still check it since its part of the API, and we never now if it may return an error in the future and cause issues. This log could help us debug issues with the container provider if it ever were to return a non-nil error. There is not need to return in that case because...
- ...`pidToCid` being a map, Go allows access to nil map, so even in case of an error, it's safe to do so. So we may be unable to use container information, but it stills allows us to do best effort in that case.